include/class.canned.php

```diff
@@ -206,6 +206,9 @@ function delete(){
         $type = array('type' => 'deleted');
         Signal::send('object.deleted', $this, $type);
 
+        // Disable Filters
+        Filter::disableFilters($this);
```

Comment:
Instead of calling `disableFiltters` allover - why not listen to `object.deleted`?